### PR TITLE
Adds various features to populates

### PR DESCRIPTION
**Pull request by javamonn**
### Description

Adds various features supporting `populates` that we found to be missing for our use-case. I've ensured the tests pass and functionally everything appears to work. We could consider splitting this PR up if some features are deemed undesired for merge.

- Adds support for lodash supported path syntax in `populates.child`, e.g. `this.is.a.nested.prop`. This was added to support populating on ids that may be nested arbitrarily in the top level object and primarily required changing direct JS prop accessors `foo[bar], foo.bar` to `_.get(foo, bar)`
- Adds support for per-item populates configurations by specifying a function callback that generates the populates configuration from the top level (key, item) tuple. This enables dynamically populating items, such as using the top level key when building the `root` path, or to only populate items passing a test when querying for a list.
- Adds support for `storeAs` when `populates` is configured to continue working around https://github.com/prescottprue/react-redux-firebase/issues/130 - is there a reason why this shouldn't work?

### Check List

- [x] All test passed
- [x] Added test to ensure to fix/ensure properly.

### CODE REVIEW

**Comment by javamonn** [comment]
Hmm, the test suite passes for me locally, will have a look later this afternoon.

**Comment by prescottprue** [comment]
These are awesome additions to populate! Going to test it out today, and look into add some stuff to the docs.

**Comment by prescottprue** [comment]
These features have been [added to the roadmap for `v1.5.0`](http://react-redux-firebase.com/docs/roadmap.html#next-minor-version-v150).

Going to mere into `v1.5.0` branch for now. Hoping to have an alpha version of `v1.5.0` soon.

File: src/actions/query.js [hunk 1]
```diff
@@ -148,7 +149,7 @@ export const watchEvent = (firebase, dispatch, { type, path, populates, queryPar
           })
           dispatch({
```
Comment (prescottprue) [inline src/actions/query.js]: This got moved to the dispatch after the populate children dispatches (keeps `isLoaded` from being `true` before all children are set to redux)

**Pull request merged**